### PR TITLE
[lld][WebAssembly] Move linker global state in to context object. NFC

### PR DESCRIPTION
**Pull request by sbc100**
See lld/ELF/Config.h

### CODE REVIEW

File: lld/wasm/Config.h [hunk 1]
```diff
@@ -96,24 +101,19 @@ struct Configuration {
...
   llvm::SmallVector<uint8_t, 0> buildIdVector;
+};
 
-  // The following config options do not directly correspond to any
-  // particular command line options, and should probably be moved to separate
-  // Ctx struct as in ELF/Config.h
+// The only instance of Configuration struct.
+extern Configuration *config;
 
+struct Ctx {
```
Comment (dschuff) [inline lld/wasm/Config.h:109]: How is `Ctx` related to `Configuration`? How do you decide what goes in here vs there?

Comment (sbc100) [reply lld/wasm/Config.h:109]: Configuration contains things that correspond to command line flags.  Ctx holds global state.   This is the same as in ELF.

Comment (dschuff) [reply lld/wasm/Config.h:109]: Makes sense. Could we maybe improve upon ELF and add a comment to that effect somewhere?

**Review by dschuff — APPROVED** [review]
otherwise LGTM

File: lld/wasm/Config.h (hunk 1 above)
Comment (sbc100) [reply lld/wasm/Config.h:109]: Done

**Comment by MaskRay** [comment]
`Ctx` is for Context. In ELF,  `Configuration ` is for user-specified options and I added `Ctx` for global variables.

**Pull request merged**